apps/sms/views/shared/js/utils.js

```diff
@@ -33,40 +33,91 @@
     },
     getFormattedHour: function ut_getFormattedHour(time) {
       this.date.shared.setTime(+time);
-      return this.date.format.localeFormat(
-        this.date.shared,
-        navigator.mozL10n.get(
-          navigator.mozHour12 ? 'shortTimeFormat12' : 'shortTimeFormat24'
-        )
-      );
+      return this.date.shared.toLocaleString(navigator.languages, {
+        hour12: navigator.mozHour12,
+        hour: 'numeric',
+        minute: 'numeric',
+      });
     },
     getDayDate: function re_getDayDate(time) {
       this.date.shared.setTime(+time);
       this.date.shared.setHours(0, 0, 0, 0);
       return this.date.shared.getTime();
     },
-    getHeaderDate: function ut_giveHeaderDate(time) {
-      var _ = navigator.mozL10n.get;
+    _getFormatter: function ut_getFormatter({options, withTime}) {
+      if (withTime) {
+        options.hour12 = navigator.mozHour12;
+        options.hour = 'numeric';
+        options.minute = 'numeric';
+      }
+      return new Intl.DateTimeFormat(navigator.languages, options);
+    },
+    setHeaderDate: function ut_setHeaderDate({time, element, withTime}) {
+      var formatter;
       var today = Utils.getDayDate(Date.now());
       var otherDay = Utils.getDayDate(time);
       var dayDiff = (today - otherDay) / 86400000;
       this.date.shared.setTime(+time);
 
       if (isNaN(dayDiff)) {
-        return _('incorrectDate');
+        element.setAttribute('data-l10n-id', 'incorrectDate');
+        return;
       }
 
       if (dayDiff < 0) {
         // future time
-        return this.date.format.localeFormat(
-          this.date.shared, '%x'
-        );
+        formatter = this._getFormatter({
+          options: {
+            month: '2-digit',
+            day: '2-digit',
+            year: 'numeric',
+          },
+          withTime: withTime
+        });
+        element.removeAttribute('data-l10n-id');
+        element.textContent = formatter.format(this.date.shared);
+        return;
       }
 
-      return dayDiff === 0 && _('today') ||
-        dayDiff === 1 && _('yesterday') ||
-        dayDiff < 6 && this.date.format.localeFormat(this.date.shared, '%A') ||
-        this.date.format.localeFormat(this.date.shared, '%x');
```

Comment:
We don't control it in Intl. It's ICU provided data (which is nice because it standardizes the patterns between systems and browsers).

I'd point out at Ecma 402 spec, but the ecma-international.org website is down :)